src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -216,13 +247,57 @@ public String ingestDataset(
     // Fill in a default load id if the caller did not provide one in the ingest request.
     String loadTag = loadService.computeLoadTag(ingestRequestModel.getLoadTag());
     ingestRequestModel.setLoadTag(loadTag);
-    String description =
-        "Ingest from "
-            + ingestRequestModel.getPath()
-            + " to "
-            + ingestRequestModel.getTable()
-            + " in dataset id "
-            + id;
+
+    String description;
+    // Note: we are writing ingested to a bucket if specified before the flight starts so that the
+    // data does not get materialized in the flight DB. This means that we also need to edit the
+    // request to remove data that shouldn't get stored
+    if (ingestRequestModel.getFormat().equals(FormatEnum.ARRAY)) {
```

Comment:
might make sense to move the logic inside the 'if formatEnum.ARRAY' case to a helper function

Reply:
Can you use IngestUtils.isIngestFromPayload here, too?